spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/MqttPahoMessageDrivenChannelAdapter.java

```diff
@@ -159,16 +159,14 @@ protected synchronized void doStop() {
 				if (this.consumerStopAction.equals(ConsumerStopAction.UNSUBSCRIBE_ALWAYS)
 						|| (this.consumerStopAction.equals(ConsumerStopAction.UNSUBSCRIBE_CLEAN)
 								&& this.cleanSession)) {
-					this.client.unsubscribe(getTopic())
-							.waitForCompletion(this.completionTimeout);
+					this.client.unsubscribe(getTopic());
 				}
 			}
 			catch (MqttException e) {
 				logger.error("Exception while unsubscribing", e);
 			}
 			try {
-				this.client.disconnect()
-						.waitForCompletion(this.completionTimeout);
+				this.client.disconnectForcibly(this.completionTimeout);
```

Comment:
You don't propagate `completionTimeout` to the `MqttClient.setTimeToWait()`.
Is that intentional?
I think so, because of that your commit message explanation.
Any way just want to be sure that we are on the same page.
Thanks